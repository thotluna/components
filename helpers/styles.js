import classNames from 'classnames'

export const getDynamicClasses = (cssModule = {}, props = {}, classes) => {
  return classes.reduce((classesObject, classKey) => {
    const propValue = props[classKey]
    const className = cssModule[`${classKey}-${propValue}`]

    return className && propValue
      ? { ...classesObject, [className]: propValue }
      : classesObject
  }, {})
}

export const getModuleClasses = (cssModule, classKey) => {
  return (cssModule && cssModule[classKey]) || classKey
}

export const getObjectClasses = (cssModule, object) => {
  return Object.keys(object).reduce((classes, classKey) => {
    const className = cssModule[classKey]
    return className ? { ...classes, [className]: object[classKey] } : classes
  }, {})
}

export const getClasses = (cssModule = {}) => (props) => (...args) => {
  const allClass = args.map((arg) => {
    if (Array.isArray(arg)) {
      const dinamicClasses = getDynamicClasses(cssModule, props, arg)
      // console.log('>>>dinamic', dinamicClasses)
      return dinamicClasses
    } else if (typeof arg === 'string') {
      const moduleClass = getModuleClasses(cssModule, arg)
      // console.log('>>>module', moduleClass)
      return moduleClass
    } else if (typeof arg === 'object') {
      const objectClass = getObjectClasses(cssModule, arg)
      // console.log('>>>object', objectClass)
      return objectClass
    }
  })
  // console.log(allClass)

  const result = classNames(allClass)

  // console.log(result)
  return result
}
