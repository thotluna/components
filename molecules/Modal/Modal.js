import React from 'react'
import PropTypes from 'prop-types'
import { styles } from '.'
import withStyles from '../../hocs/withStyles'
import Title from '../../atoms/Title'
import Buttons from '../Buttons'
import Separator from '../../atoms/SeparatorLine'
import Spacer from '../../layout/Spacer'

export const Modal = ({
  children,
  title,
  onClose,
  withFooter,
  onCancel,
  onAccept,
  getStyles,
  isPlayground,
}) => {
  return (
    <div
      className={getStyles('modal', {
        'is-playground': isPlayground,
      })}
    >
      <>
        <div className={getStyles('modal-title')}>
          <div>
            {title && (
              <Title color="primary" size="lg">
                {title}
              </Title>
            )}
          </div>
          <Buttons
            ariaLabel="button"
            color="inverter"
            icon="close"
            onCLick={onClose}
            size="short"
            withoutBorder
          />
        </div>
        <Spacer.Horizontal size="xs" />
        <Separator size="xs" title="Head finish" />
      </>

      <div className={getStyles('modal-body')}>{children}</div>
      {withFooter && (
        <>
          <Separator size="xs" title="footer" />
          <Spacer.Horizontal size="xm" />
          <div className={getStyles('modal-footer')}>
            <Buttons
              ariaLabel="button"
              color="inverter"
              onClick={onCancel}
              size="short"
            >
              Cancel
            </Buttons>
            <Buttons ariaLabel="button" onClick={onAccept} size="short">
              Aceptar
            </Buttons>
          </div>
        </>
      )}
    </div>
  )
}

Modal.propTypes = {
  children: PropTypes.node.isRequired,
  getStyles: PropTypes.func.isRequired,
  onClose: PropTypes.func,
  withFooter: PropTypes.bool,
  onCancel: PropTypes.func,
  onAccept: PropTypes.func,
  title: PropTypes.string,
  isPlayground: PropTypes.bool,
}

Modal.defaultProps = {
  withFooter: false,
  isPlayground: false,
}

export default withStyles(styles)(Modal)
