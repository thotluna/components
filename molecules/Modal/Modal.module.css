.modal {
  width: 100%;
  padding: 2em;
  background-color: var(--background-color-primary);
}

.is-playground {
  max-width: max-content;
}

.modal-title {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.modal-body {
  padding: 2em 0;
}

.modal-footer {
  display: flex;
  align-items: center;
  justify-content: flex-end;
  gap: 0.5em;
}
