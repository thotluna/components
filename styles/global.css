@import url('https://fonts.googleapis.com/css2?family=Architects+Daughter&family=Montserrat:ital,wght@0,400;0,700;1,100&display=swap');
html,
body {
  height: 100%;
  font-family: var(--font-family-title);
  font-size: var(--font-size-base);
}

#root {
  height: 100%;
}