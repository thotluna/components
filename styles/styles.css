@import './minireset.min.css';
@import './tokens.css';
@import './global.css';
@import './variables.css';
