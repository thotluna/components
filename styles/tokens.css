:root {
  --color-brand-blue-angel: #b4d1fe;
  --color-brand-light-slate-blue: #7b61ff;
  --color-brand-mercury: #e5e5e5;
  --color-brand-mid-gray: #595f68;
  --color-brand-mist-blue: #676e79;
  --color-brand-charcoal-gray: #3b3f45;
  --color-brand-blossom-ping: #feb4ef;
  --color-brand-shocking-ping: #fb0ecc;
  --color-base-black: #000;
  --color-base-white: #fff;
  --color-base-transparent: transparent;
  --spacing-none: 0;
  --spacing-xs: 5;
  --spacing-sm: 15;
  --spacing-md: 20;
  --spacing-lg: 40;
  --media-query-min-width-sm: 576px;
  --media-query-min-width-md: 768px;
  --media-query-min-width-lg: 992px;
  --media-query-min-width-xl: 1200px;
  --media-query-max-width-xs: 575.98px;
  --media-query-max-width-sm: 767.98px;
  --media-query-max-width-md: 991.98px;
  --media-query-max-width-lg: 1199px;
  --font-family-sans: 'Open Sans', arial, helvetica, sans-serif;
  --font-family-title: 'Architects Daughter', arial, helvetica, sans-serif;
  --font-size-base: 10px;
  --font-size-xs: 1.1rem;
  --font-size-sm: 1.3rem;
  --font-size-md: 1.8rem;
  --font-size-lg: 2rem;
  --font-size-xl: 2.1rem;
  --font-size-2xl: 6rem;
  --font-weight-hairline: 100;
  --font-weight-normal: 400;
  --font-weight-bold: 700;
  --line-height-none: 1;
  --line-height-tight: 1.25;
  --line-height-snug: 1.375;
  --line-height-normal: 1.5;
  --line-height-relaxed: 1.625;
  --line-height-loose: 2;
  --letter-spacing-tighter: -0.05em;
  --letter-spacing-tight: -0.025em;
  --letter-spacing-normal: 0;
  --letter-spacing-wide: 0.025em;
  --letter-spacing-wider: 0.05em;
  --letter-spacing-widest: 0.1em;
  --box-shadow-none-t: none;
  --box-shadow-xs-t: 0 3px 6px rgba(0, 0, 0, 0.05);
  --box-shadow-sm-t: 0 4px 5px rgba(0, 0, 0, 0.07);
  --border-radius-none: 0;
  --border-radius-sm: 10px;
  --border-radius-md: 20px;
  --border-radius-lg: 30px;
  --border-radius-full: 9999px;
  --border-width-none: 0;
  --border-width-thin: 1px;
  --border-width-thick: 2px;
  --opacity-0: 0;
  --opacity-25: 0.25;
  --opacity-50: 0.5;
  --opacity-75: 0.75;
  --opacity-100: 1;
  --z-index-0: 0;
  --z-index-10: 10;
  --z-index-20: 20;
  --z-index-30: 30;
  --z-index-40: 40;
  --z-index-50: 50;
  --z-index-auto: auto;
  --color-background: #3b3f45;
  --color-primary: #b4d1fe;
  --color-secondary: #feb4ef;
  --color-font-base: #b4d1fe;
  --color-font-primary: #3b3f45;
  --color-font-error: #fb0ecc;
  --button-base-background: transparent;
  --button-base-color: #b4d1fe;
  --button-base-color-border: #b4d1fe;
  --button-primary-background: #b4d1fe;
  --button-primary-color: #3b3f45;
  --button-primary-color-border: #b4d1fe;
  --button-secondary-background: #feb4ef;
  --button-secondary-color: #3b3f45;
  --button-secondary-color-border: #b4d1fe;
  --container-max-width: 800px;
}
