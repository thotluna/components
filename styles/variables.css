:root {
  /* background */
  --color-background: var(--color-dark-background);

  /* color */
  --color-base: var(--color-dark-button-base);
  --color-primary: var(--color-dark-primary);
  --color-secondary: var(--color-dark-secondary);

  /* color font */
  --color-font-base: var(--color-dark-font-base);
  --color-font-muted: var(--color-dark-font-muted);
  --color-font-primary: var(--color-font-base);

  /* Box shadow */
  --box-shadow-none: none;
  --box-shadow-xs: none;
  --box-shadow-sm: none;
  --box-shadow-focus: var(--shadow-focus-in), var(--shadow-focus-out);
  --border-radius-card: 20px;
}

@media (prefers-color-scheme: light) {
  :root {
    /* background */
    --color-background: var(--color-light-background);

    /* color */
    --color-base: var(--color-light-button-base);
    --color-primary: var(--color-light-primary);
    --color-secondary: var(--color-light-secondary);

    /* color font */
    --color-font-primary: var(--color-light-primary);
    --color-font-base: var(--color-light-font-base);
    --color-font-muted: var(--color-light-font-muted);

    /* Box shadow */
    --box-shadow-xs: var(--color-light-box-shadow-xs);
    --box-shadow-sm: var(--color-light-box-shadow-sm);
  }
}
