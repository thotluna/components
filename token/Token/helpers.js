import toKebabCase from '../../utils/toKebalCase'

export const getTokenItems = (tokens, tokenKey) => (group) => {
  if (group) {
    return Object.keys(tokens[group]).map((key) => ({
      label: `--${toKebabCase(tokenKey)}-${group}-${toKebabCase(key)}`,
      value: tokens[group][key],
    }))
  }

  return Object.keys(tokens).map((key) => ({
    label: `--${toKebabCase(tokenKey)}-${toKebabCase(key)}`,
    value: tokens[key],
  }))
}
