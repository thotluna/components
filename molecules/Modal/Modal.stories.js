import { Modal } from './Modal'
import styles from './Modal.module.css'
import { getTemplate } from '../../helpers/storybook'

const Template = getTemplate(Modal, styles)

const args = {
  children: 'Consent doubt Tuckborough challenge destroying.',
}

export default {
  title: 'Molecules/Modal',
  component: Modal,
  args,
  argTypes: {
    children: { control: 'text' },
  },
}

export const Default = Template.bind({})
Default.args = { ...args }

export const WithTitleAndFooter = Template.bind({})
WithTitleAndFooter.args = {
  ...args,
  title: 'Hey men!',
  withFooter: true,
}
