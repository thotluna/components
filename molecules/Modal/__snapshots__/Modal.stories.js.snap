// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Molecules/Modal Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="modal"
  >
    <div
      className="modal-title"
    >
      <div />
      <button
        aria-label="button"
        className="buttons size-short color-inverter without-border"
      >
        <div
          className="container color-primary size-sm"
        >
          <svg
            height={21}
            viewBox="0 0 21 21"
            width={21}
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fillRule="evenodd"
              height="100%"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              width="100%"
            >
              <path
                d="M15.5 15.5l-10-10zM15.5 5.5l-10 10"
              />
            </g>
          </svg>
        </div>
      </button>
    </div>
    <div
      className="spacer horizontal"
      style={
        Object {
          "height": 5,
        }
      }
    />
    <div
      className="container-separator color-primary"
    >
      <hr
        className="separator-line color-primary size-xs"
        title="Head finish"
      />
    </div>
    <div
      className="modal-body"
    >
      Consent doubt Tuckborough challenge destroying.
    </div>
  </div>
</div>
`;

exports[`Storyshots Molecules/Modal With Title And Footer 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="modal"
  >
    <div
      className="modal-title"
    >
      <div>
        <header
          className="headings color-primary size-xl weight-bold"
        >
          Hey men!
        </header>
      </div>
      <button
        aria-label="button"
        className="buttons size-short color-inverter without-border"
      >
        <div
          className="container color-primary size-sm"
        >
          <svg
            height={21}
            viewBox="0 0 21 21"
            width={21}
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fillRule="evenodd"
              height="100%"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              width="100%"
            >
              <path
                d="M15.5 15.5l-10-10zM15.5 5.5l-10 10"
              />
            </g>
          </svg>
        </div>
      </button>
    </div>
    <div
      className="spacer horizontal"
      style={
        Object {
          "height": 5,
        }
      }
    />
    <div
      className="container-separator color-primary"
    >
      <hr
        className="separator-line color-primary size-xs"
        title="Head finish"
      />
    </div>
    <div
      className="modal-body"
    >
      Consent doubt Tuckborough challenge destroying.
    </div>
    <div
      className="container-separator color-primary"
    >
      <hr
        className="separator-line color-primary size-xs"
        title="footer"
      />
    </div>
    <div
      className="spacer horizontal"
      style={
        Object {
          "height": undefined,
        }
      }
    />
    <div
      className="modal-footer"
    >
      <button
        aria-label="button"
        className="buttons size-short color-inverter"
      >
        <header
          className="headings color-primary size-md weight-bold"
        >
          Cancel
        </header>
      </button>
      <button
        aria-label="button"
        className="buttons size-short color-primary"
      >
        <header
          className="headings color-inverted size-md weight-bold"
        >
          Aceptar
        </header>
      </button>
    </div>
  </div>
</div>
`;
